README.md
## kylin user guide


debian/changelog
kylin-user-guide (1.0.2-71) v101; urgency=medium

  * 去掉图像查看器（eom）

 -- quyanying <quyanying@kylinos.cn>  Mon, 26 Apr 2021 13:54:22 +0800

kylin-user-guide (1.0.2-70) v101; urgency=medium

  * Fixed: bug#46545(sp1)
  * Fixed: bug#46964(990)、49715(9A0)

 -- quyanying <quyanying@kylinos.cn>  Thu, 22 Apr 2021 17:01:39 +0800

kylin-user-guide (1.0.2-69) v101; urgency=medium

  * Fixed: bug#49297

 -- guang tang <tangguang@kylinos.cn>  Mon, 19 Apr 2021 18:51:55 +0800

kylin-user-guide (1.0.2-68) v101; urgency=medium

  Feat:
    * 引入日志框架
    * 修改名称为Manual

 -- guang tang <tangguang@kylinos.cn>  Mon, 19 Apr 2021 17:14:26 +0800

kylin-user-guide (1.0.2-67) v101; urgency=medium

  * Fixed: bug#48142

 -- guang tang <tangguang@kylinos.cn>  Mon, 19 Apr 2021 14:10:51 +0800

kylin-user-guide (1.0.2-66) v101; urgency=medium

  * 合并原来的改动
  * 更新归档管理器 
  * 更新代码中软件商店的名称为software_store

 -- quyanying <quyanying@kylinos.cn>  Fri, 16 Apr 2021 16:05:36 +0800

kylin-user-guide (1.0.2-65) v101; urgency=medium

  * 添加麒麟看图更新日期

 -- zoujunnan <zoujunnan@kylinos.cn>  Thu, 15 Apr 2021 13:43:59 +0800

kylin-user-guide (1.0.2-64) v101; urgency=medium

  * 更新桌面环境中，无线网络的图片

 -- quyanying <quyanying@kylinos.cn>  Wed, 14 Apr 2021 13:43:59 +0800

kylin-user-guide (1.0.2-63) v101; urgency=medium

  * 添加麒麟安装器更新日期

 -- guang tang <tangguang@kylinos.cn>  Wed, 07 Apr 2021 10:51:49 +0800

kylin-user-guide (1.0.2-62) v101; urgency=medium

  * 把侧边栏的小插件内容移动到桌面环境里的侧边栏部分

 -- quyanying <quyanying@kylinos.cn>  Tue, 06 Apr 2021 16:22:39 +0800

kylin-user-guide (1.0.2-61) v101; urgency=medium

  * 解决切换为英文系统后，用户手册（Kylin User Guide）的显示不完全(bug#46418)

 -- guang tang <tangguang@kylinos.cn>  Fri, 02 Apr 2021 11:25:49 +0800

kylin-user-guide (1.0.2-60) v101; urgency=medium

  * 解决bug#46080、bug#43850

 -- guang tang <tangguang@kylinos.cn>  Wed, 31 Mar 2021 18:49:11 +0800

kylin-user-guide (1.0.2-59) v101; urgency=medium

  * 解决bug#45782、bug#45666、bug#44245

 -- guang tang <tangguang@kylinos.cn>  Tue, 30 Mar 2021 16:19:28 +0800

kylin-user-guide (1.0.2-58) v101; urgency=medium

  * 添加evolution邮件客户端文档时间

 -- guang tang <tangguang@kylinos.cn>  Sat, 27 Mar 2021 14:44:37 +0800

kylin-user-guide (1.0.2-57) v101; urgency=medium

  * 更新小工具-麒麟录音的图13

 -- quyanying <quyanying@kylinos.cn>  Wed, 24 Mar 2021 11:39:19 +0800

kylin-user-guide (1.0.2-56) v101; urgency=medium

  * 添加ukui-clock、ukui-notebook跳转参数

 -- guang tang <tangguang@kylinos.cn>  Wed, 24 Mar 2021 09:47:34 +0800

kylin-user-guide (1.0.2-55) v101; urgency=medium

  * 控制面板应改为设置(bug#44345)

 -- guang tang <tangguang@kylinos.cn>  Tue, 23 Mar 2021 17:06:59 +0800

kylin-user-guide (1.0.2-54) v101; urgency=medium

  * 去掉claws-mail
  * 系统安装增加两条tips，说明HW990目前仅支持直接安装模式。 

 -- quyanying <quyanying@kylinos.cn>  Mon, 22 Mar 2021 11:41:14 +0800

kylin-user-guide (1.0.2-53) v101; urgency=medium

  * 解决游戏类应用跳转手册失败

 -- tang guang <tangguang@kylinos.cn>  Sat, 20 Mar 2021 10:45:43 +0800

kylin-user-guide (1.0.2-52) v101; urgency=medium

  * 更新麒麟计算器中英文截图(bug#43456 43310)

 -- caoliang <caoliang@kylinos.cn>  Fri, 19 Mar 2021 17:43:50 +0800

kylin-user-guide (1.0.2-51) v101; urgency=medium

  * 解决菜单点击用户手册，打开失败
  * 解决游戏类应用跳转手册失败

 -- tang guang <tangguang@kylinos.cn>  Fri, 19 Mar 2021 11:25:27 +0800

kylin-user-guide (1.0.2-50) v101; urgency=medium

  * 替换主页游戏图标(bug#33264)

 -- tang guang <tangguang@kylinos.cn>  Fri, 19 Mar 2021 10:03:55 +0800

kylin-user-guide (1.0.2-49) v101; urgency=medium

  * 解决关于界面邮箱地址深色模式未反白(bug#42999)
  * 当前主题下点击左侧侧边栏n个选项，切换主题后，被点击选项会记住切换之前的主题(bug#42884)

 -- tang guang <tangguang@kylinos.cn>  Wed, 17 Mar 2021 17:27:31 +0800

kylin-user-guide (1.0.2-48) v101; urgency=medium

  * 同步github的修改记录

 -- tang guang <tangguang@kylinos.cn>  Tue, 16 Mar 2021 17:56:04 +0800

kylin-user-guide (1.0.2-47) v101; urgency=medium

  * 合并(1.0.2-46~rc1)修改

 -- tang guang <tangguang@kylinos.cn>  Tue, 16 Mar 2021 14:50:04 +0800

kylin-user-guide (1.0.2-46) v101; urgency=medium

  * 新增对help URL支持，不再依赖yelp跳转

 -- wuxiaoyi <wuxiaoyi@kylinos.cn>  Tue, 16 Mar 2021 14:14:41 +0800

kylin-user-guide (1.0.2-46~rc1) v101; urgency=medium

  * 修改软件版本号获取逻辑

 -- tang guang <tangguang@kylinos.cn>  Mon, 15 Mar 2021 16:14:19 +0800

kylin-user-guide (1.0.2-45~rc1) v101; urgency=medium

  * 解决用户手册-关于-字体没有反白处理(bug#41553)
  * 解决用户手册关于页面邮箱格式错位(bug#41367)

 -- tang guang <tangguang@kylinos.cn>  Tue, 09 Mar 2021 19:32:48 +0800

kylin-user-guide (1.0.2-44~rc2) v101; urgency=medium

  * 增加快捷键，补充桌面环境里两个快捷键
  * 更新关于里的“服务与支持团队” 

 -- quyanying <quyanying@kylinos.cn>  Mon, 08 Mar 2021 11:52:11 +0800

kylin-user-guide (1.0.2-44~rc1) v101; urgency=medium

  * 解决bug#37520、bug#37524、bug#40464

 -- tang guang <tangguang@kylinos.cn>  Wed, 03 Mar 2021 14:46:18 +0800

kylin-user-guide (1.0.2-43~rc2) v101; urgency=medium

  * 更新桌面环境、小工具的部分截图和图标

 -- quyanying <quyanying@kylinos.cn>  Mon, 22 Feb 2021 17:02:40 +0800

kylin-user-guide (1.0.2-43~rc1) v101; urgency=medium

  * 更新系统安装的图2\4\6
  * 更新小工具里麒麟录音部分，和计算器截图 

 -- quyanying <quyanying@kylinos.cn>  Sat, 06 Feb 2021 11:30:28 +0800

kylin-user-guide (1.0.2-42) v101; urgency=medium

  * （bug #38121） 【用户手册】用户手册全屏后部分图片显示过大，影响阅读体验

 -- wuxiaoyi <wuxiaoyi@kylinos.cn>  Fri, 05 Feb 2021 18:30:04 +0800

kylin-user-guide (1.0.2-41) v101; urgency=medium 38121 【用户手册】用户手册全屏后部分图片显示过大，影响阅读体验

  * 深色主题，关于窗口的关闭按钮不显示以及返回按钮看不清(bug#37864)

 -- tang guang <tangguang@kylinos.cn>  Wed, 03 Feb 2021 14:02:50 +0800

kylin-user-guide (1.0.2-40) v101; urgency=medium

  * 添加关于界面的连接跳转

 -- tang guang <tangguang@kylinos.cn>  Wed, 03 Feb 2021 11:27:43 +0800

kylin-user-guide (1.0.2-39) v101; urgency=medium

  * 同时监听任务栏和控制面板的主题切换
  * 修改部分css样式

 -- tang guang <tangguang@kylinos.cn>  Tue, 02 Feb 2021 19:59:40 +0800

kylin-user-guide (1.0.2-38) v101; urgency=medium

  * 修改版本号
  * 调整英文下技术支持板块布局(bug#37520)

 -- tang guang <tangguang@kylinos.cn>  Tue, 02 Feb 2021 09:57:03 +0800

kylin-user-guide (1.0.2-37) v101; urgency=medium

  * 添加ul标签在深色模式下的样式

 -- tang guang <tangguang@kylinos.cn>  Mon, 01 Feb 2021 20:43:19 +0800


src/aboutwidget.cpp
#include "aboutwidget.h"
#include "xatom-helper.h"

AboutWidget::AboutWidget(QWidget *parent):
    QDialog(parent)
{
    this->setFixedSize(420,360);

    if(QGSettings::isSchemaInstalled("org.ukui.style")){
        settings = new QGSettings("org.ukui.style");
        connect(settings,&QGSettings::changed,this,&AboutWidget::gsettingsChange);
    }

    // 添加窗管协议
    MotifWmHints hints;
    hints.flags = MWM_HINTS_FUNCTIONS|MWM_HINTS_DECORATIONS;
    hints.functions = MWM_FUNC_ALL;
    hints.decorations = MWM_DECOR_BORDER;
    XAtomHelper::getInstance()->setWindowMotifHint(this->winId(), hints);

    initUI();
}

AboutWidget::~AboutWidget()
{

}

void AboutWidget::initUI()
{
    title_icon = new QLabel(this);
    title_icon->setGeometry(8,8,24,24);

    title_name = new QLabel(this);
    title_name->setGeometry(40,8,300,25);
    title_name->setAlignment(Qt::AlignLeft|Qt::AlignVCenter);

    close_btn = new QPushButton(this);
    close_btn->setIcon(QIcon::fromTheme("window-close-symbolic"));
    close_btn->setProperty("isWindowButton", 0x2);
    close_btn->setProperty("useIconHighlightEffect", 0x8);
    close_btn->setFlat(true);
    close_btn->setFocusPolicy(Qt::NoFocus);
    close_btn->setToolTip(tr("Close"));
    close_btn->setCheckable(false);
//    close_btn->setCursor(QCursor(Qt::ArrowCursor));
    close_btn->setGeometry(386,4,30,30);
    connect(close_btn,&QPushButton::clicked,this,&AboutWidget::close);

    app_icon = new QLabel(this);
    app_icon->setGeometry(162,72,96,96);

    QFont font;
    font.setPixelSize(18);

    app_name = new QLabel(this);
    app_name->setGeometry(10,184,400,28);
    app_name->setAlignment(Qt::AlignCenter);
    app_name->setFont(font);

    font.setPixelSize(14);
    app_version = new QLabel(this);
    app_version->setGeometry(10,224,400,24);
    app_version->setAlignment(Qt::AlignCenter);
    app_version->setFont(font);

//    palette.setColor(QPalette::WindowText,QColor(89,89,89));
//    app_version->setPalette(palette);

    app_description = new QLabel(this);
    app_description->setGeometry(32,260,356,30);
    app_description->setAlignment(Qt::AlignTop|Qt::AlignHCenter);
    app_description->setWordWrap(true);

    app_support = new QLabel(this);
    app_support->setGeometry(32,300,356,30);
    app_support->setAlignment(Qt::AlignHCenter);
    app_support->setOpenExternalLinks(true);

    QPalette palette;
    if(settings->get("styleName").toString() == "ukui-black"|| settings->get("styleName").toString() == "ukui-dark"){
        palette.setBrush(QPalette::Background, QBrush(QColor(29,29,31)));
        palette.setBrush(QPalette::Text, QBrush(QColor(Qt::white)));
        app_support->setText(tr("Service & Support : <a style='color: white;' href='mailto://support@kylinos.cn'>support@kylinos.cn</a>"));
    }else{
        palette.setBrush(QPalette::Background, QBrush(QColor(Qt::white)));
        palette.setBrush(QPalette::Text, QBrush(QColor(29,29,31)));
        app_support->setText(tr("Service & Support : <a style='color: black;' href='mailto://support@kylinos.cn'>support@kylinos.cn</a>"));
    }
    this->setPalette(palette);

}

void AboutWidget::setAppIcon(const QString &text)
{
    if(text.isEmpty() || text == ""){
        qDebug() << Q_FUNC_INFO << "arg is empty!!!";
        return;
    }else{
        if(QIcon::hasThemeIcon(text)){
            title_icon->setPixmap(QIcon::fromTheme(text).pixmap(24,24));
            app_icon->setPixmap(QIcon::fromTheme(text).pixmap(96,96));
        }else{
            qDebug() << Q_FUNC_INFO << "No icon exists: " << text << " !!!";
        }
    }
}

void AboutWidget::setAppName(const QString &text)
{
    if(text.isEmpty() || text == ""){
        qDebug() << Q_FUNC_INFO << "APP name should not be empty!!!";
        return;
    }else{
        title_name->setText(text);
        title_name->update();
        app_name->setText(text);
    }
}

void AboutWidget::setAppVersion(const QString &text)
{
    if(text.isEmpty() || text == ""){
        qDebug() << Q_FUNC_INFO << "APP version should not be empty!!!";
        return;
    }else{
        app_version->setText(tr("VERSION")+":  "+text);
    }
}

void AboutWidget::setAppDescription(const QString &text)
{
    app_description->setText(text);
}

void AboutWidget::gsettingsChange(const QString &key)
{
    QPalette palette;
    if(key == "styleName"){
        if(settings->get("styleName").toString() == "ukui-black"|| settings->get("styleName").toString() == "ukui-dark"){
            palette.setBrush(QPalette::Background, QBrush(QColor(29,29,31)));
            palette.setBrush(QPalette::Text, QBrush(QColor(Qt::white)));
            app_support->setText(tr("Service & Support : <a style='color: white;' href='mailto://support@kylinos.cn'>support@kylinos.cn</a>"));
        }else{
            palette.setBrush(QPalette::Background, QBrush(QColor(Qt::white)));
            palette.setBrush(QPalette::Text, QBrush(QColor(29,29,31)));
            app_support->setText(tr("Service & Support : <a style='color: black;' href='mailto://support@kylinos.cn'>support@kylinos.cn</a>"));
        }
    }
    this->setPalette(palette);
}

//void AboutWidget::closeEvent(QCloseEvent *event)
//{
//    Q_UNUSED(event);
//    qDebug() << Q_FUNC_INFO;
//}
